src/Stormpath/DataStore/DefaultDataStore.php

```diff
@@ -97,9 +103,18 @@ public function getResource($href, $className, array $options = array())
         }
 
         $queryString = $this->getQueryString($options);
+
+
         $data = $this->executeRequest(Request::METHOD_GET, $href, '', $queryString);
 
-        return $this->resourceFactory->instantiate($className, array($data, $queryString));
+
+        if($this->resourceIsCacheable($data)) {
+            $this->addDataToCache($data);
+        }
```

Comment:
Where are you attempting to get the resource data from the cache before executing the request to the server?